metadata_backend/api/parser.py

```diff
@@ -0,0 +1,239 @@
+"""Tool to parse XML files to JSON."""
+
+import random
+from pprint import pprint
+from typing import Dict, List
+
+from aiohttp import web
+from xmlschema import XMLSchema, XMLSchemaException
+
+from ..helpers.schema_load import SchemaLoader, SchemaNotFoundException
+
+
+class SubmissionXMLToJSONParser:
+    """Methods to parse necessary data from different xml types.
+
+    Currently only submission-type is parsed excplicitly, others are parsed
+    just by flattening them.
+    """
+
+    def __init__(self):
+        """Create SchemaLoader instance for loading schemas."""
+        self.loader = SchemaLoader()
+
+    def load_schema(self, xml_type: str) -> XMLSchema:
+        """Load schema for validation and xml-to-json decoding.
+
+        :param xml_type: Schema to be loaded
+        :returns: Schema instance matching the given schema
+        :raises: HTTPBadRequest if schema wasn't found
+        """
+        try:
+            schema = self.loader.get_schema(xml_type)
+        except (SchemaNotFoundException, XMLSchemaException) as error:
+            reason = f"{error} {xml_type}"
+            raise web.HTTPBadRequest(reason=reason)
+        return schema
+
+    @staticmethod
+    def sort_actions_by_schemas(data: List) -> List:
+        """Sort schemas according to their priority.
+
+        Schemas need to be processed in certain order (for example 'study'
+        submission needs to processed first in order to generate accession
+        number for other submissions.
+
+        Sorting order is based on ENA Metadata Model:
+        https://ena-docs.readthedocs.io/en/latest/submit/general-guide/metadata.html
+
+        This should probably be refactored later, since submissions can be also
+        made via front-end or with actions in POST-request (i.e. without
+        submission.xml).
+
+        :param data: Data to be sorted
+        :returns: Sorted list
+        """
+        order = {"study": 1, "sample": 2, "experiment": 3, "run": 4,
+                 "analysis": 5, "dac": 6, "policy": 7, "dataset": 8,
+                 "project": 9}
+        return sorted(data, key=lambda x: order[x["schema"]])
+
+    @staticmethod
+    def flatten_and_parse_json(data):
+        """Recursively flatten json and parse away unwanted characters.
+
+        :param data: JSON to be flattened
+        :returns: Flattened JSON
+        """
+        out = {}
+
+        def flatten(content, name=""):
```

Comment:
maybe a miss also here?
```suggestion
        def flatten(content: Union[Dict, List], name: str = "") -> None:
```
`Union` needs to be imported from typing